lib/passwords.ts

```diff
@@ -88,9 +88,14 @@ export interface StrengthCheckResponse extends BaseResponse {
 export interface MigrateRequest {
   email: string;
   hash: string;
-  hash_type: "bcrypt";
+  hash_type: string;
```

Comment:
The typescript idiomatic way of doing this is to do a union type

```
interface MigrateRequestBase {
  email: string;
  hash: string;
}

interface ArgonMigrateRequest extends MigrateRequestBase {
  hash_type: 'argon2';
  argon_2_salt?: string;
  argon_2_iteration_amount?: string;
  argon_2_memory?: string;
  argon_2_threads?: string;
  argon_2_key_length?: string;
}

interface BcryptMigrateRequest extends MigrateRequestBase {
   hash_type: "bcrypt";
}

interface OtherMigrateRequest extends MigrateRequestBase {
  hash_type: 'whatever';
  whatever_param?: whatever;
}

export interface MigrateRequest = BCryptMigrateRequest | OtherMigrateRequest | ArgonMigrateRequest
```

and that way hash-type specific params will only be valid/allowed if that hash type is selected. It also allows you to make a param required for a specific type without making it required for all types

Reply:
https://github.com/stytchauth/stytch-node/pull/151/commits/f5517567b5398f503681515b1884759620bf34c7